packages/synapse-react-client/src/components/widgets/facet-nav/PlotsContainer.tsx

```diff
@@ -85,6 +72,62 @@ const generatePlotKey = (plotUiState: UiPlotState) => {
   }
 }
 
+const isPlotInState = (
+  plotId: PlotIdentifier,
+  plotUiStateArray: UiPlotState[],
+): boolean => {
+  return !!plotUiStateArray.find(plot => {
+    return plotMatchesDefinition(plot.plotId, plotId)
+  })
+}
+
+const getCustomPlotIdentifier = (
+  customPlot: QueryWrapperSynapsePlotProps,
+): PlotIdentifier => {
+  return { __custom: true, title: customPlot.title ?? '' }
+}
+
+const getCombinedNewPlots = (
+  customPlots: QueryWrapperSynapsePlotProps[],
+  facetNavPanelPropsArray: Pick<
+    FacetNavPanelProps,
+    'applyChangesToFacetFilter' | 'applyChangesToGraphSlice' | 'facetToPlot'
+  >[],
+): UiPlotState[] => [
+  ...(customPlots ?? []).map((plotProps, index) => ({
+    plotId: getCustomPlotIdentifier(plotProps),
+    isHidden: index >= DEFAULT_VISIBLE_PLOTS,
+    plotType: convertPlotlyPlotTypeToFacetNavPlotType(plotProps.type),
+  })),
+  ...(facetNavPanelPropsArray ?? []).map((facetPlotProps, index) => ({
+    plotId: facetPlotProps.facetToPlot,
+    isHidden: index + (customPlots ?? []).length >= DEFAULT_VISIBLE_PLOTS,
+    plotType: DEFAULT_PLOT_TYPE,
+  })),
```

Comment:
We can make this a little more simple with default arguments
```suggestion
  customPlots: QueryWrapperSynapsePlotProps[] = [],
  facetNavPanelPropsArray: Pick<
    FacetNavPanelProps,
    'applyChangesToFacetFilter' | 'applyChangesToGraphSlice' | 'facetToPlot'
  >[] = [],
): UiPlotState[] => [
  ...customPlots.map((plotProps, index) => ({
    plotId: getCustomPlotIdentifier(plotProps),
    isHidden: index >= DEFAULT_VISIBLE_PLOTS,
    plotType: convertPlotlyPlotTypeToFacetNavPlotType(plotProps.type),
  })),
  ...facetNavPanelPropsArray.map((facetPlotProps, index) => ({
    plotId: facetPlotProps.facetToPlot,
    isHidden: index + customPlots.length >= DEFAULT_VISIBLE_PLOTS,
    plotType: DEFAULT_PLOT_TYPE,
  })),
```